routes/users.js

```diff
@@ -1,14 +1,13 @@
 const router = require('express').Router();
-const { getUsers, getUserById, createUser, updateUser, updateUserAvatar, deleteUser } = require('../controllers/users');
+const {
+  getUsers, getUserById, updateUser, updateUserAvatar,
+} = require('../controllers/users');
+const { checkUserObjectId, checkUserObjectIdAndParamId } = require('../middlewares/checkIf');
 
 router.get('/', getUsers);
-router.get('/:id', getUserById);
+router.get('/:id', checkUserObjectId, getUserById);
 
-router.post('/', createUser);
-
-router.patch('/:id', updateUser);
-router.patch('/:id/avatar', updateUserAvatar);
-
-router.delete('/:id', deleteUser);
+router.patch('/:id', checkUserObjectIdAndParamId, updateUser);
+router.patch('/:id/avatar', checkUserObjectIdAndParamId, updateUserAvatar);
```

Comment:
Нужно исправить: роуты не соответствуют заданию.
В целом обработка внутри ок, все работает. Только по заданию должны быть:
- `PATCH /users/me`,
- `PATCH /users/me/avatar`

```suggestion
router.patch('/me', checkUserObjectIdAndParamId, updateUser);
router.patch('/me/avatar', checkUserObjectIdAndParamId, updateUserAvatar);
```

Идея такая, что пользователь может обновлять только свой профиль. Иначе требовалось бы создавать поле, для определения пользователей с расширенными правами.

Я вроде писал об этом ранее, но мог пропустить 😭 .

P.S. нужно не забыть поправить и в описании:
https://github.com/dimukko/mesto-backend/pull/3/files#diff-04c6e90faac2675aa89e2176d2eec7d8R84-R91

Reply:
Всё нормально, вы писали об этом, это я считал, что под /me подразумевается как раз наш текущий id :))) Исправлю сейчас, спасибо, дошло :))